beacon-chain/blockchain/process_block.go

```diff
@@ -501,6 +501,14 @@ func (s *Service) handleEpochBoundary(ctx context.Context, postState state.Beaco
 		if err := helpers.UpdateProposerIndicesInCache(ctx, copied); err != nil {
 			return err
 		}
+
+		if err := copied.SetSlot(copied.Slot() + params.BeaconConfig().SlotsPerEpoch); err != nil {
```

Comment:
I don't like this. This is a hack that changes the state's slot by 32. after adding this the state becomes invalid in an inconsistent status and this invalid state is sent down the path to another core function. This in itself I think is very much error prone as any changes ito `UpdateProposerIndicesInCache` will need to remember that it will be getting an invalid state when it arrives. If it's only this, it's simpler to change the signature of that function to also receive the slot we want to update for. 

In fact from the form of that function it seems to me that it's simpler to pass the root since we already have it in slot 31. In the case that 31 is skipped we can do this on `lateBlockTasks` with the latest stateroot as well.